common-ktx/src/main/java/co/nimblehq/common/extensions/ViewExt.kt

```diff
@@ -0,0 +1,142 @@
+package co.nimblehq.common.extensions
+
+import android.graphics.*
+import android.view.View
+import androidx.annotation.DimenRes
+import androidx.annotation.Px
+
+/**
+ * Make view visible.
+ */
+fun View.visible() {
+    this.visibility = View.VISIBLE
+}
+
+/**
+ * Make view invisible.
+ */
+fun View.invisible() {
+    this.visibility = View.INVISIBLE
+}
+
+/**
+ * Make view gone.
+ */
+fun View.gone() {
+    this.visibility = View.GONE
+}
+
+/**
+ * Check if the visible condition is true; set view visible; otherwise, set view gone.
+ *
+ * @param visible The condition to make view visible.
+ */
+fun View.visibleOrGone(visible: Boolean) {
+    if (visible) visible() else gone()
+}
+
+/**
+ * Check if the visible condition is true; set view visible; otherwise, set view gone.
+ * The visibility will active after the delay.
+ *
+ * @param visible The condition to make view visible.
+ * @param delay The delay time in millisecond to affect the view.
+ */
+fun View.visibleOrGone(visible: Boolean, delay: Long) {
+    if (delay <= 0L) {
+        visibleOrGone(visible)
+    } else {
+        postDelayed({ visibleOrGone(visible) }, delay)
```

Comment:
@lydiasama do we need to check if the context of view is still valid?

Reply:
@doannimble Did you mention something like this?
https://stackoverflow.com/a/64535290

Reply:
@lydiasama Exactly!